pysmt/smtlib/parser/parser.py

```diff
@@ -610,7 +610,7 @@ def _get_quantified_var(self, name, type_name):
             return self._get_var(name, type_name)
         except PysmtTypeError:
             return self.env.formula_manager.FreshSymbol(typename=type_name,
```

Comment:
I think that it would be better to always return a FreshSymbol with a 'strange' name, so that the original name is kept free

Reply:
A test that should highlight this issue is something like:
```
(define-fun f ((n Int)) Int n)
(declare-fun n () Real)
```

Reply:
Regarding the FreshSymbol - did you mean doing this in _get_quantifed_var in general? Or just for the case of (define-fun)? I have pushed the former, which results in failures like [this](https://travis-ci.org/yoni206/pysmt/jobs/395520821) one:
```
  File "/home/travis/build/yoni206/pysmt/pysmt/test/smtlib/test_parser_examples.py", line 50, in test_parse_examples
    self.assertEqual(f_in.simplify(), f_out.simplify())
AssertionError: (forall __y0 . (x -> __y0)) != (forall y . (x -> y))
```

Reply:
What about:

'''
  File "/home/travis/build/yoni206/pysmt/pysmt/test/test_regressions.py", line 396, in test_smtlib_define_fun_serialization
    self.assertEqual(res, smtlib_input)
AssertionError: '(define-fun init ((__x0 Bool)) Bool (and __x0 (and __x0 (and[43 chars]))))' != '(define-fun init ((x Bool)) Bool (and x (and x (and x (and x[19 chars]))))'
- (define-fun init ((__x0 Bool)) Bool (and __x0 (and __x0 (and __x0 (and __x0 (and __x0 (and __x0 __x0)))))))
?                    -- -                  -- -      -- -      -- -      -- -      -- -      -- ^^^^ -
+ (define-fun init ((x Bool)) Bool (and x (and x (and x (and x (and x (and x x)))))))
?                                                                           ^
'''

I guess this is expected as we always generate a fresh symbol.
But is this a problem with the test or with the fix?

Reply:
I see. I think we can accommodate the test as this is expected behavior now.

It's not great, but something like:
```python
self.assertEqual(res.replace('__x0', 'x'), smtlib_input)
```
should work.